.github/workflows/actionlint.yml

```diff
@@ -5,6 +5,11 @@ on:
       - '.github/**'
     types: [opened, synchronize, reopened, ready_for_review]
 
+# cancel existing runs of the same workflow on the same ref
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.ref }}
```

Comment:
We're using `${{ github.head_ref || github.run_id }}-diff` elsewhere, and I confess I don't understand the tradeoffs with your approach.  Should we update those to match this, or vice versa?

Reply:
That's a good question, it depends.  Using the workflow and the ref ensures that a new run for the same ref cancels any in-progress of the same workflow on the same ref.  In particular, this targets cases where a branch is updated several times in quick succession (fixing typos, fixing lint errors, etc.).

Using the ref and a constant like `-diff` allows the control to effect multiple workflows but is still specific to the ref (usually a branch).  So if you have exactly one workflow using the `$ref-diff` concurrency group, you get the same effect as if you tied it to the workflow name.  But if you add another workflow that uses the `$ref-diff` group, you get different behavior -- a new run of `workflow-1` using `$ref-diff` could cancel an in-progress run of `workflow-1` or it could cancel an in-progress run of `workflow-2` if that also uses `$ref-diff`.

I think workflow+ref is the right default behavior, but `$ref-diff` is not wrong unless something breaks.  So there's no need to update the others unless you foresee future changes where they would break.  Conversely, I'm happy to update any of my additions if you think they should be consolidated.

Reply:
Cool, ty!